arrow-optics/src/main/kotlin/arrow/optics/Fold.kt

```diff
@@ -34,94 +16,95 @@ inline fun <S, A> FoldOf<S, A>.fix(): Fold<S, A> = this as Fold<S, A>
  * @param S the source of a [Fold]
  * @param A the target of a [Fold]
  */
-interface Fold<S, A> : FoldOf<S, A> {
+interface Fold<S, A> {
 
   /**
    * Map each target to a type R and use a Monoid to fold the results
    */
-  fun <R> foldMap(M: Monoid<R>, s: S, f: (A) -> R): R
-
-  companion object {
-
-    fun <A> id() = PIso.id<A>().asFold()
-
-    /**
-     * [Fold] that takes either [S] or [S] and strips the choice of [S].
-     */
-    fun <S> codiagonal() = object : Fold<Either<S, S>, S> {
-      override fun <R> foldMap(M: Monoid<R>, s: Either<S, S>, f: (S) -> R): R = s.fold(f, f)
-    }
-
-    /**
-     * Creates a [Fold] based on a predicate of the source [S]
-     */
-    fun <S> select(p: (S) -> Boolean): Fold<S, S> = object : Fold<S, S> {
-      override fun <R> foldMap(M: Monoid<R>, s: S, f: (S) -> R): R = if (p(s)) f(s) else M.empty()
-    }
-
-    /**
-     * [Fold] that points to nothing
-     */
-    fun <A, B> void() = POptional.void<A, B>().asFold()
-
-    /**
-     * Create a [Fold] from a [arrow.Foldable]
-     */
-    fun <F, S> fromFoldable(foldable: Foldable<F>) = object : Fold<Kind<F, S>, S> {
-      override fun <R> foldMap(M: Monoid<R>, s: Kind<F, S>, f: (S) -> R): R = foldable.run { s.foldMap(M, f) }
-    }
-  }
+  fun <R> foldMap(M: Monoid<R>, source: S, f: (focus: A) -> R): R
```

Comment:
I didn't know you could name the input parameter of a function type :)

Reply:
Yes, it's super useful actually and it's also taking into account by IDEA to suggest names for lambda params. We should do this in all lambdas.. but since it's not a breaking change (also not binary breaking), we can even do this gradually after 1.0 afaik.